### PR TITLE
Add impl for UserPresence

### PR DESCRIPTION
**Pull request by kofls**
cc: @kaczmarczyck

### CODE REVIEW

File: examples/rust/opensk/src/user_presence.rs [hunk 1]
```diff
@@ -12,22 +12,75 @@
...
+                button::Pressed => button_pressed.set(true),
+                button::Released => (),
+            };
+            let listener = Listener::new(index, handler);
+            listener.leak();
+        }
+
+        // Start a periodic timer blinking the LED.
+        // TODO: Similarly, should we be blinking all LEDs or only index 0?
+        let blinking = Rc::new(Cell::new(false));
```
Comment (ia0) [inline examples/rust/opensk/src/user_presence.rs:53]: We don't need this variable (and you probably meant to initialize it to true). It is enough to do:

```rust
let blink_timer = clock::Timer::new(|| led::set(0, !led::get(0)));
led::set(0, led::On);
blink_timer.start(Periodic, Duration::from_millis(200));
// [...]
scheduling::wait_until(|| button_pressed.get() || timeout.get());
drop(blink_timer);
led::set(0, led::Off);
```

**Review by ia0 — COMMENTED** [review]
Looks good modulo the 2 comments.

File: examples/rust/opensk/src/user_presence.rs [hunk 2]
```diff
@@ -12,22 +12,75 @@
...
+        // TODO: Should we be checking for any button pressed or only index 0?
+        let button_pressed = Rc::new(Cell::new(false));
+        for index in 0 .. num_buttons {
+            let button_pressed = button_pressed.clone();
+            let handler = move |state| match state {
+                button::Pressed => button_pressed.set(true),
+                button::Released => (),
+            };
+            let listener = Listener::new(index, handler);
+            listener.leak();
```
[the remark this answers is not in the document]
Comment (kofls) [reply examples/rust/opensk/src/user_presence.rs]: Ok, got it. Removed `leak()`

File: examples/rust/opensk/src/user_presence.rs (hunk 1 above)
Comment (kofls) [reply examples/rust/opensk/src/user_presence.rs]: Done.

**Review by ia0 — APPROVED** [review]

**Pull request merged**